apps/studio/prisma/schema.prisma

```diff
@@ -71,16 +71,21 @@ model Resource {
 
   createdAt DateTime @default(now())
   updatedAt DateTime @default(now()) @updatedAt
+
+  @@unique([siteId, permalink], name: "unique_site_permalink")
```

Comment:
caveat here is that this means within the same siteId, the permalink must be unique. But what we are ensuring is the full qualified permalink - which might be expensive to re-create recursively, but given that create/edit page settings is a relatively less frequent operation.

Am okay with this for now, but is something we should take note of